src/net.rs

```diff
@@ -746,15 +874,97 @@ impl NetworkManager {
         rx_packets.clear();
         if let Some(chat_messages) = rx_chat_messages {
             chat_messages.clear();
+            chat_messages.buffer_wrap_index = None;
         }
     }
+
+    #[allow(unused)]
+    pub fn print_statistics(&self) {
+        info!("Tx Successes: {}", self.statistics.tx_packets_success);
+        info!("Tx Failures:  {}", self.statistics.tx_packets_failed);
+    }
+
+    #[allow(unused)]
+    pub fn retransmit_expired_tx_packets(&mut self,
+         udp_tx: &mpsc::UnboundedSender<(SocketAddr, Packet)>,
+         addr: SocketAddr,
+         confirmed_ack: Option<u64>,
+         indices: &Vec<usize>) {
+
+        let mut error_occurred = false;
+        let mut failed_index = 0;
+
+        // Retransmit all packets after that are still in the queue after RETRANSMISSION_THRESHOLD_IN_MS
+        for index in indices.iter() {
```

Comment:
It's a little clearer to me to use `for &index in indices.iter()` and then just refer to it in the block as `index` instead of `*index`.